specification/synapse/resource-manager/Microsoft.Synapse/preview/2019-06-01-preview/sqlPool.json

```diff
@@ -3833,6 +3833,174 @@
+            "in": "path",
+            "name": "sqlPoolName",
+            "description": "SQL pool name",
+            "required": true
+          },
+          {
+            "name": "restorePointName",
+            "in": "path",
+            "description": "The name of the restore point.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "consumes": [],
+        "produces": [
+          "application/json"
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully returned the restore point.",
+            "schema": {
+              "$ref": "#/definitions/RestorePoint"
+            }
+          },
+          "default": {
+            "description": "*** Error Responses: ***\n\n * 400 FeatureDisabledOnSelectedEdition - User attempted to use a feature which is disabled on current database edition.\n\n * 400 BackupNotAllowedOnDeactivatedDatabase - Cannot create restore point on a deactivated database.\n\n * 400 RestorePointLimitReached - A restore point cannot be created because database would exceed the allowed quota of restore points.\n\n * 400 RestorePointAttemptToDeleteSystemBackup - Cannot delete system restore point.\n\n * 404 DatabaseRestorePointNotFound - Can not find database restore point.\n\n * 404 SubscriptionDoesNotHaveServer - The requested server was not found\n\n * 404 DatabaseDoesNotExist - User has specified a database name that does not exist on this server instance.\n\n * 408 DatabaseRestorePointTimedOut - Create database restore point failed."
+          }
+        },
+        "x-ms-examples": {
+          "Gets a Sql pool restore point.": {
+            "$ref": "./examples/SqlPoolRestorePointsGet.json"
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "SqlPools, RestorePoints"
+        ],
+        "description": "Deletes a restore point.",
+        "operationId": "RestorePoints_Delete",
+        "parameters": [
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v1/types.json#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "./workspace.json#/parameters/workspaceNameParameter"
+          },
+          {
+            "type": "string",
+            "in": "path",
+            "name": "sqlPoolName",
+            "description": "SQL pool name",
+            "required": true
+          },
+          {
+            "name": "restorePointName",
+            "in": "path",
+            "description": "The name of the restore point.",
+            "required": true,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
```

Comment:
If you are purely proxying these through then you know what restore points the user has created via your APIs and can return 204 appropriately correct?

Reply:
Are you suggesting we detect the present of content, and return 204 is content is absent, and 200 otherwise ?

Reply:
Yes.  A 200 should only be returned if something was actually deleted.  SQL has been around forever and has a bunch of bad, old legacy behavior in their APIs.  We shouldn't carry that forward

Reply:
For legacy reasons we will stay this way for now.